src/wallet/rpcwallet.cpp

```diff
@@ -2186,16 +2186,24 @@ UniValue walletpassphrase(const UniValue& params, bool fHelp)
     // Note that the walletpassphrase is stored in params[0] which is not mlock()ed
     SecureString strWalletPass;
     strWalletPass.reserve(100);
-    // Get rid of this .c_str() by implementing SecureString::operator=(std::string)
-    // Alternately, find a way to make params[0] mlock()'d to begin with.
-    strWalletPass = params[0].get_str().c_str();
+    strWalletPass = std::string_view{params[0].get_str()};
 
     if (strWalletPass.length() > 0)
     {
         LOCK2(cs_main, pwalletMain->cs_wallet);
 
         if (!pwalletMain->Unlock(strWalletPass))
-            throw JSONRPCError(RPC_WALLET_PASSPHRASE_INCORRECT, "Error: The wallet passphrase entered was incorrect.");
+            // Check if the passphrase has a null character (see #27067 for details)
```

Comment:
Maybe clarify that it references an issue on the Bitcoin repo.

Reply:
Ah I missed this, thanks